src/battle-dex.ts

```diff
@@ -539,21 +541,31 @@ const Dex = new class implements ModdedDex {
 			facing = 'back';
 		}
 
-		// Decide what gen sprites to use.
-		let fieldGenNum = options.gen;
-		if (Dex.prefs('nopastgens')) fieldGenNum = 6;
-		if (Dex.prefs('bwgfx') && fieldGenNum >= 6) fieldGenNum = 5;
-		let genNum = Math.max(fieldGenNum, Math.min(template.gen, 5));
-		let gen = ['', 'gen1', 'gen2', 'gen3', 'gen4', 'gen5', '', ''][genNum];
+		// Decide which gen sprites to use.
+		//
+		// There are several different generations we care about here:
+		//
+		//   - mechanicsGen: the generation number of the mechanics and battle (options.gen)
+		//   - graphicsGen: the generation number of sprite/field graphics the user has requested.
+		//     This will default to mechanicsGen, but may be altered depending on user preferences.
+		//   - spriteData.gen: the generation number of a the specific Pokemon sprite in question.
+		//     This defaults to graphicsGen, but if the graphicsGen doesn't have a sprite for the Pokemon
+		//     (eg. Darmanitan in graphicsGen 2) then we go up gens until it exists.
+		//
+		let graphicsGen = mechanicsGen;
+		if (Dex.prefs('nopastgens')) graphicsGen = 6;
+		if (Dex.prefs('bwgfx') && graphicsGen >= 6) graphicsGen = 5;
+		spriteData.gen = Math.max(graphicsGen, Math.min(template.gen, 5));
+		const baseDir = ['', 'gen1', 'gen2', 'gen3', 'gen4', 'gen5', '', ''][spriteData.gen];
```

Comment:
I renamed this as well. `gen` is too confusing when theres already 3 other gens at play and its not actually always a generation in the case of `''`.